### PR TITLE
Added demo code for a Classifier and Transformer

### PR DESCRIPTION
**Pull request by vighneshbirodkar**
Addresses #10 
The code for the 1-NN classifier was taken from 
https://github.com/scikit-learn/scikit-learn/issues/6526

/ping @MechCoder (my best friend)
@mblondel

### CODE REVIEW

File: skltemplate/template.py [hunk 1]
```diff
@@ -47,4 +51,135 @@ def predict(self, X):
...
+            The target values. An array of int.
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        # Check that X and y have correct shape
+        X, y = check_X_y(X, y)
+        # Store the classes seen durind fit
+        self.classes_ = unique_labels(y)
```
Comment (MechCoder) [inline skltemplate/template.py:92]: Using `check_classification_targets` from utils.multiclass should be a better way of testing this, especially since it was made for this specific purpose.

File: examples/plot_classifier.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+X_test = rng.rand(500, 2)
+y_pred = clf.predict(X_test)
+
+X_0 = X_test[y_pred == 0]
+X_1 = X_test[y_pred == 1]
+
+ax0 = plt.scatter(X_0[:, 0], X_0[:, 1], c='crimson', s=30)
+ax1 = plt.scatter(X_1[:, 0], X_1[:, 1], c='deepskyblue', s=30)
+
+
```
Comment (MechCoder) [inline examples/plot_classifier.py:40]: 1 extra line

File: examples/plot_classifier.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+rng = np.random.RandomState(13)
+X_test = rng.rand(500, 2)
+y_pred = clf.predict(X_test)
+
+X_0 = X_test[y_pred == 0]
+X_1 = X_test[y_pred == 1]
+
+ax0 = plt.scatter(X_0[:, 0], X_0[:, 1], c='crimson', s=30)
+ax1 = plt.scatter(X_1[:, 0], X_1[:, 1], c='deepskyblue', s=30)
+
```
Comment (MechCoder) [inline examples/plot_classifier.py:25]: Eventhough it is a dummy example, it might be good to just plot the points [0, 0] and [1, 1]

File: examples/plot_transformer.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+"""
+import numpy as np
+from skltemplate import TemplateTransformer
+from matplotlib import pyplot as plt
+
+X = np.arange(50).reshape(-1, 1)
+estimator = TemplateTransformer()
+X_transformed = estimator.fit_transform(X)
+
+plt.plot(X.flatten()/X.max(), label='Original Data')
```
Comment (MechCoder) [inline examples/plot_transformer.py]: space after `/`

File: examples/plot_transformer.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+plt.plot(X_transformed.flatten()/X_transformed.max(), label='Transformed Data')
+plt.title('Scaled plots of original and transformed data')
```
Comment (MechCoder) [inline examples/plot_transformer.py]: Labels to the axes?

**Comment by MechCoder** [comment]
@vighneshbirodkar Thanks! Just some minor cosmetics. Feel free to merge after addressing.

Do you think it would be a good idea to document somewhere what has to be done so that tests pass after inheriting from `ClassifierMixin` and `TransformerMixin`

To me it seems for a classifier, we need to make sure that the target is discrete, fitting on a single label should return the same label during predict. For a transformer, fit_transform should be consistent with fit().transform() etc..

File: skltemplate/template.py (hunk 1 above)
Comment (vighneshbirodkar) [inline skltemplate/template.py:92]: At least in this case we can't do anything with multiple labels.

**Comment by vighneshbirodkar** [comment]
@MechCoder Can you have one more look ? I have addressed all except one comment. I have added comments in the code as to why each step is necessary. What other documentation you think is necessary ?

File: skltemplate/template.py (hunk 1 above)
Comment (MechCoder) [inline skltemplate/template.py:92]: I agree that `check_classification_targets` is placed in the wrong module

**Comment by MechCoder** [comment]
Thanks and sorry for the delay
